lib/datadog/tracing/contrib/rack/header_tagging.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
```

Comment:
This code was moved here from `lib/datadog/tracing/contrib/sinatra/env.rb` and `lib/datadog/tracing/contrib/sinatra/headers.rb`.
Sinatra already had helpers to perform header matching for Rack-style headers.
Rack was also performing the matching, but in a less organized fashion.

Because the `DD_TRACE_HEADER_TAGS` option has to be added to both Rack and Sinatra, this PR merges both header matching implementations into this file.